core/mempool/src/lib.rs

```diff
@@ -255,23 +255,27 @@ where
         tx_hashes: Vec<Hash>,
     ) -> ProtocolResult<Vec<SignedTransaction>> {
         let len = tx_hashes.len();
+        let mut missing_hashes = vec![];
         let mut full_txs = Vec::with_capacity(len);
 
-        for tx_hash in &tx_hashes {
+        for tx_hash in tx_hashes.iter() {
```

Comment:
It's impossible that the query transactions in both memory and storage. So this modification can be confusing.

Reply:
This requires you to get the lock first (flush_lock), but `pull_txs` is an externally accessible interface and is accessed very frequently, and getting a mutex each time can significantly block the process.